lifecycle-operator/apis/lifecycle/v1alpha4/keptnworkloadversion_types.go

```diff
@@ -0,0 +1,470 @@
+}
+
+func (w KeptnWorkloadVersion) GetNamespace() string {
+	return w.Namespace
+}
+
+func (w *KeptnWorkloadVersion) SetState(state common.KeptnState) {
+	w.Status.Status = state
+}
+
+func (w KeptnWorkloadVersion) GetCurrentPhase() string {
+	return w.Status.CurrentPhase
+}
+
+func (w *KeptnWorkloadVersion) SetCurrentPhase(phase string) {
+	w.Status.CurrentPhase = phase
+}
+
+func (w KeptnWorkloadVersion) GetVersion() string {
+	return w.Spec.Version
+}
+
+func (w KeptnWorkloadVersion) GenerateTask(taskDefinition v1alpha3.KeptnTaskDefinition, checkType common.CheckType) v1alpha3.KeptnTask {
+	return v1alpha3.KeptnTask{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        common.GenerateTaskName(checkType, taskDefinition.Name),
+			Namespace:   w.Namespace,
+			Labels:      taskDefinition.Labels,
+			Annotations: taskDefinition.Annotations,
+		},
+		Spec: v1alpha3.KeptnTaskSpec{
+			Context: v1alpha3.TaskContext{
+				WorkloadName:    w.GetParentName(),
+				AppName:         w.GetAppName(),
+				WorkloadVersion: w.GetVersion(),
+				TaskType:        string(checkType),
+				ObjectType:      "Workload",
+			},
+			TaskDefinition:   taskDefinition.Name,
+			Parameters:       v1alpha3.TaskParameters{},
+			SecureParameters: v1alpha3.SecureParameters{},
+			Type:             checkType,
+			Retries:          taskDefinition.Spec.Retries,
+			Timeout:          taskDefinition.Spec.Timeout,
+		},
+	}
+}
+
+func (w KeptnWorkloadVersion) GenerateEvaluation(evaluationDefinition v1alpha3.KeptnEvaluationDefinition, checkType common.CheckType) v1alpha3.KeptnEvaluation {
+	return v1alpha3.KeptnEvaluation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      common.GenerateEvaluationName(checkType, evaluationDefinition.Name),
+			Namespace: w.Namespace,
+		},
+		Spec: v1alpha3.KeptnEvaluationSpec{
+			AppName:              w.GetAppName(),
+			WorkloadVersion:      w.GetVersion(),
+			Workload:             w.GetParentName(),
+			EvaluationDefinition: evaluationDefinition.Name,
+			Type:                 checkType,
+			RetryInterval: metav1.Duration{
+				Duration: 5 * time.Second,
+			},
+		},
+	}
+}
+
+func (w KeptnWorkloadVersion) GetSpanAttributes() []attribute.KeyValue {
+	return []attribute.KeyValue{
+		common.AppName.String(w.Spec.AppName),
+		common.WorkloadName.String(w.Spec.WorkloadName),
+		common.WorkloadVersion.String(w.Spec.Version),
+		common.WorkloadNamespace.String(w.Namespace),
+	}
+}
+
+func (w KeptnWorkloadVersion) GetSpanKey(phase string) string {
+	return fmt.Sprintf("%s.%s.%s.%s", w.Spec.TraceId["traceparent"], w.Spec.WorkloadName, w.Spec.Version, phase)
+}
+
+func (w KeptnWorkloadVersion) GetSpanName(phase string) string {
+	if phase == "" {
+		return w.Name
+	}
+	return fmt.Sprintf("%s/%s", w.Spec.WorkloadName, phase)
+}
+
+func (w KeptnWorkloadVersion) SetSpanAttributes(span trace.Span) {
+	span.SetAttributes(w.GetSpanAttributes()...)
+}
+
+//nolint:dupl
+func (w *KeptnWorkloadVersion) DeprecateRemainingPhases(phase common.KeptnPhaseType) {
```

Comment:
do we still need this logic now that we have a new type? 🤔

Reply:
yes, it will be used in controller (adaptations of controllers will be part of following PR)

Reply:
edit: not in controller, but in phaseHandler